openprescribing/dmd2/management/commands/import_dmd2.py

```diff
@@ -311,3 +313,166 @@ def set_vmp_bnf_codes(self):
+        #           u'Voltarol 2.32% gel'
+        #       },
+        #       AMPP: {
+        #           u'Voltarol 2.32% gel (GSK) 100 gram',
+        #           u'Voltarol 2.32% gel (GSK) 30 gram',
+        #           u'Voltarol 2.32% gel (GSK) 50 gram',
+        #       },
+        #   }
+        #   '1003020U0AAAIAI': {...},
+        #   ...
+        #   }
+
+        bnf_code_to_cls_to_names = {}
+
+        for cls in VMP, AMP, VMPP, AMPP:
+            for obj in cls.objects.all():
+                if obj.bnf_code is None:
+                    continue
+                if obj.bnf_code not in bnf_code_to_cls_to_names:
+                    bnf_code_to_cls_to_names[obj.bnf_code] = {
+                        VMP: set(),
+                        AMP: set(),
+                        VMPP: set(),
+                        AMPP: set(),
+                    }
+                cls_to_names = bnf_code_to_cls_to_names[obj.bnf_code]
+                cls_to_names[type(obj)].add(obj.nm)
+
+        # For each presentation, we find all VMPs will that presentation's BNF
+        # code.  If there is one VMP, we take its name and assign that to the
+        # presentation's dmd_name field.  If there are multiple VMPs, we try to
+        # infer a name using get_common_name().  If no such VMPs exist, we look
+        # at AMPs, then VMPPs, then AMPPs.
+
+        for presentation in Presentation.objects.all():
+            try:
+                cls_to_names = bnf_code_to_cls_to_names[presentation.bnf_code]
+            except KeyError:
+                continue
+
+            for cls in VMP, AMP, VMPP, AMPP:
+                names = cls_to_names[cls]
+                if len(names) > 0:
+                    common_name = get_common_name(list(names))
+                    if common_name is not None:
+                        presentation.dmd_name = common_name
+                        presentation.save()
+                    break
+
+
+def get_common_name(names):
+    '''Find left substring common to all names, by splitting names on spaces,
+    and possibly ignoring certain common words.
+
+    >>> get_common_name([
+        'Polyfield Soft Vinyl Patient Pack with small gloves',
+        'Polyfield Soft Vinyl Patient Pack with medium gloves',
+        'Polyfield Soft Vinyl Patient Pack with large gloves',
+    ])
+    'Polyfield Soft Vinyl Patient Pack'
+
+    Returns None if left substring is less than half the length (in words) of
+    the first name.
+
+    This function could be improved.  For instance, in dm+d there are several
+    cases where names of two products with the same BNF code differ by one word
+    in the middle:
+
+    * Imodium 2mg capsules
+    * Imodium Original 2mg capsules
+
+    And there are several cases where names of two products with the same BNF
+    code have different versions of the same ratio:
+
+    * Taurolidine 5g/250ml intraperitoneal solution bottles
+    * Taurolidine 2g/100ml intraperitoneal solution bottles
```

Comment:
This is probably one we should fix. Currently it would return "Taurolidine", right? Which is just wrong. 

As I understand it, there are two VMPPs where the pack size is a 250ml or 100ml bottle. 

We should probably normalise to some standard ratio. 

In this case the MDR name is `Taurolidine_IntraperitonealSoln100mg/5ml`

Rich has suggested going for %w/v which means g of solute/100 mL of solution so this would be `Taurolidine intraperitoneal solution 2% w/v` - but see discussion [here]( https://ebmdatalab.slack.com/archives/C31D62X5X/p1556529092090200) (specifically, getting Brian to concur!)

Reply:
> Currently it would return "Taurolidine", right?

No, because "Taurolidine" is too short.  So it won't infer a name at all.